Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
+{
+	/**
+	 * @param int $parentId
+	 * @return void
+	 */
+	public function setParentId($parentId);
+	
+	/**
+	 * @return int
+	 */
+	public function getParentId();
+	
+	/**
+	 * @param int $questionId
+	 * @return void
+	 */
+	public function setId($questionId);
```

Comment:
IMO this is highly dangerous on a public interface, as it allows consumers to mess with ids. This would be another case for the repository pattern.

Reply:
For my understanding the data gateway or a kind of repository is to be used internaly, a question is registered in the consumers using IDs. These IDs are to be given to the asq service in any kind.

When you delete a question in the pool what would you expect in the checkboxes of the questions table? We need to use the IDs there.

Other services also get IDs passed. Imagine the Taxonomy service or the Competence service.